app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -52,26 +53,34 @@ def destroy
   end
 
   def deploy_all
-    @stages = Project.all.flat_map do |project|
-      environment = Environment.find(params[:environment_id]) if params[:environment_id]
+    environment = deploy_group.environment
+    deploys = deploy_group.stages.map do |stage|
+      template_stage = Stage.where(project: stage.project, environment: environment, is_template: true).first
 
-      stages = stages_in_same_environment(project, environment)
-      next unless deploy = stages.map(&:last_successful_deploy).compact.sort_by(&:created_at).last
-      stages.map { |s| [s, deploy] }
-    end.compact
-  end
+      last_success_deploy = template_stage.last_successful_deploy
+      next unless last_success_deploy
 
-  def deploy_all_now
```

Comment:
We no longer need the interstitial page, as we won't be selecting and environment anymore, and will be deploying everything listed on the Deploy controller `show.html.erb` page.